src/m365/aad/commands/group/group-user-list.ts

```diff
@@ -0,0 +1,152 @@
+import { User } from '@microsoft/microsoft-graph-types';
+import { Logger } from '../../../../cli/Logger.js';
+import GlobalOptions from '../../../../GlobalOptions.js';
+import { aadGroup } from '../../../../utils/aadGroup.js';
+import { odata } from '../../../../utils/odata.js';
+import { validation } from '../../../../utils/validation.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  groupId?: string;
+  groupDisplayName?: string;
+  role?: string;
+}
+
+class AadGroupUserListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.GROUP_USER_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists users of a specific Azure AD group';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'displayName', 'userPrincipalName', 'userType'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupDisplayName: typeof args.options.groupDisplayName !== 'undefined',
+        role: typeof args.options.role !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: "-i, --groupId [groupId]"
+      },
+      {
+        option: "-n, --groupDisplayName [groupDisplayName]"
+      },
+      {
+        option: "-r, --role [type]",
+        autocomplete: ["Owner", "Member", "Guest"]
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['groupId', 'groupDisplayName']
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.role) {
+          if (['Owner', 'Member', 'Guest'].indexOf(args.options.role) === -1) {
+            return `${args.options.role} is not a valid role value. Allowed values Owner|Member|Guest`;
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const groupId = await this.getGroupId(args.options);
+      let users = await this.getOwners(groupId, logger);
+
+      if (args.options.role !== 'Owner') {
+        const membersAndGuests = await this.getMembersAndGuests(groupId, logger);
+        users = users.concat(membersAndGuests);
```

Comment:
Another thing: I believe users can be part of the owners list as well as of the members list. Doing a concat would add them twice in this scenario.